pkg/clustertree/cluster-manager/utils/leaf_model_handler.go

```diff
@@ -301,3 +310,91 @@ func NewLeafModelHandler(cluster *kosmosv1alpha1.Cluster, rootClientset, leafCli
 	}
 	return classificationModel
 }
+
+func (h ClassificationHandler) getClusterHealthStatus(leafClient kubernetes.Interface) (online, healthy bool) {
+	healthStatus, err := healthEndpointCheck(leafClient, "/readyz")
+	if err != nil && healthStatus == http.StatusNotFound {
+		healthStatus, err := healthEndpointCheck(leafClient, "/healthz")
+		if err != nil {
+			klog.Errorf("Failed to healthEndpointCheck healthStatus is %v, err is : %v ", healthStatus, err)
+		}
+	}
+	if err != nil {
+		klog.Errorf("Failed to do cluster health check for cluster %v, err is : %v ", h.Cluster.Name, err)
+		return false, false
+	}
+	if healthStatus != http.StatusOK {
+		klog.Infof("Member cluster %v isn't healthy", h.Cluster.Name)
+		return true, false
+	}
+
+	return true, true
+}
+
+func healthEndpointCheck(client kubernetes.Interface, path string) (int, error) {
+	var healthStatus int
+	resp := client.Discovery().RESTClient().Get().AbsPath(path).Do(context.TODO()).StatusCode(&healthStatus)
+	return healthStatus, resp.Error()
+}
+
+func getObservedReadyStatus(online, healthy bool) []corev1.NodeCondition {
+	if !online {
+		return []corev1.NodeCondition{
+			{
+				Type:    corev1.NodeReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  "ClusterNotReachable",
+				Message: "cluster is not reachable.",
+			},
+		}
+	}
+	if !healthy {
+		return []corev1.NodeCondition{
+			{
+				Type:    corev1.NodeReady,
+				Status:  corev1.ConditionFalse,
```

Comment:
```suggestion
				Status:  corev1.ConditionTrue,
```
The unhealthy member cluster does not necessarily mean that the nodes are in a not-ready state.